File: 백준/1325.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
vector<vector<int>>graph;
vector <bool> visited;
int cnt;
void dfs(int v) {
	visited[v] = true;
	for (int i = 0; i < graph[v].size(); i++) {
		int next = graph[v][i];
		if (!visited[next]) {
			cnt++;
			dfs(next);
		}
	}
}
int main() {
	int N, M;
	cin >> N >> M;
	graph.resize(N+1);
	int max = 0;
	vector <int> answer;
	for (int i = 0; i < M; i++) {
		int a, b;
		cin >> a >> b;
    //단방향 그래프
		graph[b].push_back(a);
	}
	for (int i = 1; i <= N; i++) {
		cnt = 0;
		visited.assign(N + 1, false);
		dfs(i);
		if (cnt > max) {
			answer.clear();
			answer.push_back(i);
			max = cnt;
		}
		else if (cnt == max) {
			answer.push_back(i);
		}
	}
	sort(answer.begin(), answer.end());
	for (int i = 0; i < answer.size(); i++) {
		cout << answer[i] << " ";
	}
}

File: 백준/1325.js
const [NM, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M]=NM.split(' ').map(Number);
const arr=input.map((e)=>e.split(' ').map(Number))
const graph=Array.from(Array(Number(N+1)), () => new Array());
const maxComputer=Array(N+1).fill(0);

const bfs=(v)=>{
    const visited=Array(N+1).fill(false);
    let stack=[];
    let cnt=0;
    stack.push(v);
    visited[v]=true;
    
    while(stack.length) {
        const current=stack.shift();
        for (let i=0; i<graph[current].length; i++) {
            const next=graph[current][i];
            if (!visited[next]) {
                visited[next]=true;
                stack.push(next);
                cnt++;
            }
        }
    }
    maxComputer[v]=cnt;
    
}

for (const [a, b] of arr) {
    graph[b].push(a);
}
for (let i=1; i<=N; i++) {
    bfs(i);
}

const answer=[];
const maxHack=Math.max.apply(null, maxComputer);

for (let i=1; i<=N; i++) {
    if (maxComputer[i]==maxHack) answer.push(i);
}


console.log(...answer);
